dask_ml/model_selection/_hyperband.py

```diff
@@ -137,59 +161,68 @@ def _model_id(s, n_i):
+            assert job['iterations'] > 0
             i = rng.choice(len(X_train))
-            model_future = model_futures[job["model_id"]]
-            model_future = client.submit(_partial_fit,
-                                         model_future,
-                                         X_train[i],
-                                         y_train[i],
-                                         meta=job, dry_run=dry_run,
-                                         id=job['model_id'],
-                                         fit_params=fit_params)
-
-            score_future = client.submit(_score, model_future, X_test, y_test,
-                                         dry_run=dry_run)
-            model_futures[job["model_id"]] = model_future
+
+            # This block prevents communication of the model
+            model_meta_future = model_meta_futures[job["model_id"]]
+            model_meta_future = client.submit(_partial_fit, model_meta_future,
+                                              X_train[i], y_train[i],
+                                              meta=job, dry_run=dry_run,
+                                              id=job['model_id'],
+                                              fit_params=fit_params)
+            model_meta_futures[job["model_id"]] = model_meta_future
+
+            score_future = client.submit(_score, model_meta_future, X_test,
+                                         y_test, dry_run=dry_run)
             seq.add(score_future)
 
-    return list(completed_jobs.values()), model_futures
+    assert completed_jobs.keys() == model_meta_futures.keys()
+    return list(completed_jobs.values()), params, model_meta_futures
 
 
 class HyperbandCV(DaskBaseSearchCV):
-    def __init__(self, model, params, max_iter=81, eta=3, random_state=42,
-                 scoring=None, iid=True, cv=2, cache_cv=False, **kwargs):
+    def __init__(self, model, params, max_iter=81, eta=3, asynchronous=None,
+                 random_state=42, scoring=None, iid=True, cv=1, cache_cv=False,
+                 **kwargs):
         self.model = model
         self.params = params
         self.max_iter = max_iter
         self.eta = eta
         self.random_state = random_state
         self.scoring = scoring
+        if asynchronous is None:
+            asynchronous = False
+        self.asynchronous = asynchronous
+
+
         if not iid:
             raise ValueError('Please specify iid=True. Hyperband assumes that '
                              'each observation is independent and '
                              'identitically distributed because it trains on '
                              'each block of X and y')
-        if cv != 2:
-            raise ValueError('Please specify cv=2. Future work is to '
+        if cv != 1:
+            raise ValueError('Please specify cv=1. Future work is to '
                              'allow arbitrary cross-validation (and pull '
                              'request welcome!).')
```

Comment:
I recommend the following:

    raise NotImplementedError("Cross validation is not yet Implemented.  Only cv=1 is supported")

Notes like pull requets welcome are of course true, but we probably shouldn't say this at all of Dask's exceptions, and we should be consistent across the project.

Reply:
Alternatively, maybe its better just to not include this parameter for now?  That way we won't have to do any deprecation work if we change the default value in the future.

Reply:
Good point. But the values for `cache_cv`, `iid` and `cv` are assumed in the current Hyperband implementation. Should we enforce that or only mention it in the documentation?

Reply:
> Good point. But the values for cache_cv, iid and cv are assumed in the current Hyperband implementation

You're saying that the code below assumes that, for example, `cv==1`?  If so then yes, I think that we should only mention it in documentation.  I think that we should only include parameters if the user might reasonably change them.  I don't know enough about the `cache_cv` parameter to make an intelligent statement there.

There might be a counter-argument if these are standard among scikit-learn estimators and part of an expected API.  In that case then I don't know what's best.

Reply:
> we should only include parameters if the user might reasonably change them.

I've arrived at the same conclusion separately – I was getting distracted by accepting `**kwargs` in `__init__` (and now it's removed).